addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -81,18 +96,62 @@ class ComposerTextInput extends Component {
             textInputContent: this.getContent(),
             textInputCursorStart: this._getSelectionStart(),
             textInputCursorEnd: this._getSelectionEnd(),
+            mentionedPartners: this._getMentions(),
         });
     }
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
-        /**
+    /**
+     * Detects if mentions suggestions should be displayed when user is typing
+     * and searches partners based on user's research
+     *
+     * @private
+     */
+    async _detectDelimiter() {
+        this._mentionWord = this._validateKeyword('@', false);
+        if (this._mentionWord !== false) {
+            await this._mentionFetchThrottled(
+                'res.partner',
+                'get_mention_suggestions',
+                {
+                    limit: 3,
+                    search: this._mentionWord,
+                }
+            );
+
+            if (this.state.mentionSuggestions[0].length === 0) {
+                this.state.isMentionsOpen = false;
+                this.state.active_id = 0;
+            } else {
+                this.state.isMentionsOpen = true;
+                this.state.active_id = this.state.mentionSuggestions[0][0].id;
+            }
```

Comment:
I feel like fetch of mentions + data on suggestions should be on the store (i.e. in `mail.messaging.entity.Composer`).
Something like a relation between composer and partners for suggestions (one2many, list of suggested partners, can use getter for special order based on keyword), and another for "actual" mentions (also one2many, here don't care of order).